relays/lib-substrate-relay/src/finality/target.rs

```diff
@@ -108,13 +108,17 @@ impl<P: SubstrateFinalitySyncPipeline> TargetClient<FinalitySyncPipelineAdapter<
 		header: SyncHeader<HeaderOf<P::SourceChain>>,
 		mut proof: SubstrateFinalityProof<P>,
 	) -> Result<Self::TransactionTracker, Error> {
-		// runtime module at target chain may require optimized finality proof
-		P::FinalityEngine::optimize_proof(&self.client, &header, &mut proof).await?;
+		// verify and runtime module at target chain may require optimized finality proof
+		let current_set_id =
```

Comment:
Nit: Is this a leftover ? Shouldn't it be something like  `finality_verification_context` ?